traits/trait_handlers.py

```diff
@@ -2757,6 +2757,7 @@ def update ( self, value ):
         if not hasattr(self, 'trait'):
             return set.update(self, value)
         try:
+            value = set(value)
```

Comment:
It is a performance worry. This adds computational overhead to the most common use case. We recognize this now and can fix it easily.